nexus/types/src/deployment/blueprint_diff.rs

```diff
@@ -138,6 +797,28 @@ pub struct BpDiffZonesModified {
     pub zones: Vec<ModifiedZone>,
 }
 
+impl BpDiffZonesModified {
+    pub fn new<'a>(
+        generation_before: Generation,
+        generation_after: Generation,
+        zone_diffs: impl Iterator<Item = &'a BlueprintZoneConfigDiff<'a>>,
+    ) -> (BpDiffZonesModified, BpDiffZoneErrors) {
```

Comment:
I don't love this interface, but I'm not sure what to suggest instead. Do you think an enum with 3 cases might be clearer? Something like

```rust
enum BpDiffZonesResult {
    Ok(BpDiffZonesModified), // all comparisons are fine
    Err(BpDiffZoneErrors), // all comparisons failed
    Mixed(BpDiffZonesModified, BpDiffZoneErrors), // some of each
}
```

since we expect to basically always be in the `Ok(_)` case?

Reply:
Eh, matching that in callers seems worse.